source/SemVerAnalyzer/Engine/Semver.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Pushpay.SemVerAnalyzer.Abstractions;
+
+namespace Pushpay.SemVerAnalyzer.Engine
+{
+	public class Semver : IComparable<Semver>
+	{
+		public int Major { get;set; }
+		public int Minor { get;set; }
+		public int Patch { get;set; }
+		public string Prerelease { get;set; }
+		public string Trailer { get;set; }
+
+		public Semver(int major, int minor, int patch, string prerelease, string trailer){
+			Major = major;
+			Minor = minor;
+			Patch = patch;
+			Prerelease = prerelease;
+			Trailer = trailer;
+		}
+
+		public override string ToString() => $"{Major}.{Minor}.{Patch}{(Prerelease != null ? Prerelease : "")}{(Trailer != null ? Trailer : "")}";
```

Comment:
You don't have to null check strings in an interpolated string
```suggestion
		public override string ToString() => $"{Major}.{Minor}.{Patch}{Prerelease}{Trailer}";
```

Reply:
I beat you to it 😛 but you have a committable fix.